### ISSUE TITLE
consult about pretrain

### ISSUE THREAD

**Issue by mafangniu** [mli0603/openpi-comet#14]
Hi, I am interested in reproducing or understanding the training process for the pi05-b1kpt50-cs32 model. I noticed there is a configuration named pi05_b1k-pt50_cs32_bs64_lr2.5e-5_step50k in src/openpi/training/config.py.

Could you please provide some details regarding the training of this specific model?

1. Configuration Consistency: Is the training configuration used for the released pi05-b1kpt50-cs32 model exactly consistent with the pi05_b1k-pt50_cs32_bs64_lr2.5e-5_step50k entry in config.py?

2. Hardware: What hardware devices were used for training (e.g., type and number of GPUs)?

3. Training Duration & Epochs: Approximately how many epochs does the 50k steps correspond to, and how long did the training take in total?

Thank you for your help!

**Comment by DelinQu** [comment]
Hi! Thanks for your interest in reproducing / understanding the training pipeline for pi05-b1kpt50-cs32 — happy to clarify.

> 1. What the config name means (pi05_b1k-pt50_cs32_bs64_lr2.5e-5_step50k)

- pt50: uses data covering all 50 BEHAVIOR-1K tasks
- cs32: chunk size = 32
- bs64: per-GPU batch size = 64
- step50k: per-GPU steps = 50k

The config `pi05_b1k-pt50_cs32_bs64_lr2.5e-5_step50k` corresponds to our base pretraining setup trained from official BEHAVIOR demonstrations + motion-planner-generated trajectories (CuRobo-based augmentation). What we released in the repo is closer to our weighted post-training / mixture setting (resume from base model), and the most representative configs are: `pi05-b1k-pt12-demo0_6-comet0_4-step20k` and `pi05-b1k-pt50-demo0_6-comet0_4-step20k`. Conceptually: (official demo) + (comet-1.5k RFT data) with a weighted mixture (e.g., demo0.6 / comet0.4).

> 2. Hardware used

Training was run on NVIDIA H200 GPUs.

> 3 Training duration, steps → epochs, and total cost. Using pi05_b1k-pt50_cs32_bs64_lr2.5e-5_step50k as the reference:

Due to time constraints, we did not finish the full 50k steps. In practice, we trained ~20k steps, which is roughly ~3 epochs over the corresponding training mixture (approximation; exact epoch count depends on dataloader details such as filtering, sampling weights, and any rebalancing). Total compute cost for this run is approximately ~19k H200 GPU-hours (normalized estimate; wall-clock time depends on the number of GPUs used).

**Comment by mafangniu** [comment]
Hi again — I have two follow-up questions:

For the base model pretraining (e.g., pi05_b1k-pt50_cs32_bs64_lr2.5e-5_step50k), roughly how many H200 GPUs were used (and was it multi-node)?

When I estimate the effective number of epochs, I’m using something like, is OK?:

<img width="738" height="96" alt="Image" src="https://github.com/user-attachments/assets/8d5c2b70-3976-4261-82e1-983fd5a04c1b" />

**Variables:**
- **train_steps**: Total training steps
- **batch_per_GPU**: Batch size per GPU
- **num_GPUs**: Number of GPUs
- **total_frames_in_training_dataset**: Total frames in the dataset
- **epochs**: Number of training epochs

**Comment by cn-hezhu** [comment]
Hello, I have a few questions regarding reproducing the pre-trained model. 
I followed the configuration in src/openpi/training/config.py for pi05_b1k-pt12_cs32_bs64_lr2.5e-5_step50k, setting fine_grained_level to 0, and trained it on an 8-GPU A100. During testing, the success rate for some tasks was close to 0, showing a significant difference from the open-source checkpoint (openpi_comet/pi05-b1kpt12-cs32) results. 
I'm also curious about the number of H200 GPUs you used during training. 
Is the large difference in results related to the batch size or fine_grained_level? 
Will you be releasing orchestrators-related data in the future?
